core/spec/models/spree/preferences/preferable_spec.rb

```diff
@@ -301,6 +316,7 @@ def self.allowed_admin_form_preference_types
                      default: 'my_default_secret',
                      encryption_key: 'VkYp3s6v9y$B?E(H+MbQeThWmZq4t7w!'
 
+        @a = A.new
```

Comment:
If this is not here and the test is not the first to run in the context the test will fail.

And I do no understand why it's necessary...

Reply:
Hey, @Roddoric, did you find out why that's needed?

Reply:
No, I'm not looking into it right now
I'll try to look into it during the weekend or next week 😬

Reply:
Sure! Let us know your findings 🙏

Reply:
Okk

I think I found the problem...
Context:
- `class A` is declared once in a `before(:all)` so the class WILL NOT be destroyed/recreate for each test
- `@a = A.new` is in a `before(:each)` therefore the variable is initialized before the test start
- The 4 examples from l.285 define the same preference `:secret` but the last one is the only using a default

If the test run first:
- `@a` is instantiated without `preference :secret`
- `@a.preferences` looks like `{:color=>"green"}`
- We define the `preference :secret` with a default
- Calling `@a.get_preference(:secret)` will work because it will look for the default 

If the test doesn't run first:
- `@a` is instantiated with a `preference :secret` without `:default`
- `@a.preferences` looks like `{:color=>"green", :secret=>"rJ0s--jPH2d2iyyB9KseLU--HbYfXFtRB36+yj9L3fSDFA=="}
` (`:secret` value is `nil` but encrypted)
- We override `preference :secret` with a default
- `@a.get_preference(:secret)` returns `nil` as it's initialized and does not look for the default


Setting `@a = A.new` after the override works because it'll be initialized with the new default

So there is multiple ways to fix it:

1. Defining `class A` and `class B` in a `before(:each)` instead of a `before(:all)` which will make the tests truly independents
2. Using different names for the secrets which will avoid conflict
3. Changing `@a` and `@b` initialization from:
```ruby
  before :each do
    @a = A.new
    @b = B.new
  end
```
To:
```ruby
  let(:a) { A.new }
  let(:b) { B.new }
```
It will initialize the variable once called (after the override)

IMO I would go for 1 + 3 (as 2 looks more like a workaround)